src/utils/number_with_commas.ts

```diff
@@ -0,0 +1,16 @@
+/**
+ * Adds commas to longer integers for a more visually appealing display
+ *
+ * @throws on non-integer values represented as a decimal
+ *
+ * @example
+ *
+ * const num = numberWithCommas(10_000_000);
+ * console.log(num); // result is '10,000,000'
+ */
+export default function numberWithCommas(x: number): string {
+	if (!Number.isInteger(x)) {
+		throw new Error('Value must be a non-decimal integer to add commas!');
+	}
+	return x.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ',');
```

Comment:
Looks quite a bit like the solution from SO here: https://stackoverflow.com/questions/2901102/how-to-print-a-number-with-commas-as-thousands-separators-in-javascript

A few things:
1) It's a good idea to leave behind a link to the SO solution you used in order to provide context for readers.
2) There's a good solution in there that uses `toLocaleString()` which may actually provide more readable outputs for users based on their locale settings.

Re: point 2 - I know I just riffed the "let's add commas" requirement out of thin air, but in retrospect, honoring locale settings seems like it might be a better option.